stan/math/rev/fun/to_vector.hpp

```diff
@@ -0,0 +1,29 @@
+#ifndef STAN_MATH_REV_FUN_TO_VECTOR_HPP
+#define STAN_MATH_REV_FUN_TO_VECTOR_HPP
+
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/rev/meta.hpp>
+#include <stan/math/rev/core.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Cast a `var_value<Matrix>` to a `var_value<ColumnVector>`.
+ * @tparam EigMat Inner type of the `var_value` that must inherit from
+ *  `Eigen::EigenBase`.
+ * @param x A var whose inner matrix type is to be cast to an Column matrix.
+ * @return A view of the original `x` with inner value and adjoint matrices
+ *  mapped to a column vector.
```

Comment:
[optional] I think it is not clear what exactly casting means for complex types that can not actually be typecast one into another. I would call this operation reshaping.

Reply:
Reasonable, fixed!